jobs/base_workers.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package jobs
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+type SimpleWorker struct {
+	name      string
+	stop      chan bool
+	stopped   chan bool
+	jobs      chan model.Job
+	jobServer *JobServer
+	execute   func(job *model.Job) error
+	isEnabled func(cfg *model.Config) bool
+}
+
+func NewSimpleWorker(name string, jobServer *JobServer, execute func(job *model.Job) error, isEnabled func(cfg *model.Config) bool) *SimpleWorker {
```

Comment:
I think I prefer to keep it somehow like is now, and maybe in the future move this to independent packages, and receive some kind of interface instead of an explicit jobServer. I think in the future we are going to have more information to know what needs to be coupled and what can be decoupled. For now I think the current balance is just fine until we figure out the final shape of the jobs (I expect to have at least 2 o 3 more iterations over it until we have something in really good shape).